addons/pos_restaurant/static/src/app/floor_screen/floor_screen.js

```diff
@@ -616,6 +635,30 @@ export class FloorScreen extends Component {
     getChildren(table) {
         return this.pos.models["restaurant.table"].filter((t) => t.parent_id?.id === table.id);
     }
+    async uploadImage(event) {
+        const file = event.target.files[0];
+        if (!file.type.match(/image.*/)) {
+            this.dialog.add(AlertDialog, {
+                title: _t("Unsupported File Format"),
+                body: _t("Only web-compatible Image formats such as .png or .jpeg are supported."),
+            });
+        } else {
+            const imageUrl = await getDataURLFromFile(file);
+            const loadedImage = await _loadImage(imageUrl);
+            if (loadedImage) {
+                this.floor_background_image = imageUrl;
+                this.pos.data.write("restaurant.floor", [this.activeFloor.id], {
+                    floor_background_image: imageUrl.split(",")[1],
+                });
+            } else {
+                this.dialog.add(AlertDialog, {
+                    title: _t("Loading Image Error"),
+                    body: _t("Encountered error when loading image. Please try again."),
+                });
+            }
+            this.state.isColorPicker = false;
+        }
+    }
```

Comment:
It seems that the server process the image in a different way, reload the image from the server after upload it.

```suggestion
    async uploadImage(event) {
        const file = event.target.files[0];
        if (!file.type.match(/image.*/)) {
            this.dialog.add(AlertDialog, {
                title: _t("Unsupported File Format"),
                body: _t("Only web-compatible Image formats such as .png or .jpeg are supported."),
            });
        } else {
            const imageUrl = await getDataURLFromFile(file);
            const loadedImage = await _loadImage(imageUrl);
            if (loadedImage) {
                await this.pos.data.ormWrite("restaurant.floor", [this.activeFloor.id], {
                    floor_background_image: imageUrl.split(",")[1],
                });
                await this.pos.data.read("restaurant.floor", [this.activeFloor.id]);
            } else {
                this.dialog.add(AlertDialog, {
                    title: _t("Loading Image Error"),
                    body: _t("Encountered error when loading image. Please try again."),
                });
            }
            this.state.isColorPicker = false;
        }
    }
```